src/main/java/cceclipseplugin/core/PluginManager.java

```diff
@@ -49,30 +61,124 @@ public static PluginManager getInstance() {
+			}
+		});
+
+		new Thread(() -> {
+			try {
+				wsManager.connect();
+			} catch (ConnectException e) {
+				e.printStackTrace();
+			}
+		}).start();
+
 		registerNotificationHooks();
 
-		// Start listeners
+		// Start editor & document listeners
 		PlatformUI.getWorkbench().getDisplay().asyncExec(new Runnable() {
 			@Override
 			public void run() {
-				editorChangeListener = EditorChangeListener.getInstance();
+				editorChangeListener = new EditorChangeListener();
 				PlatformUI.getWorkbench().getActiveWorkbenchWindow().getPartService()
 						.addPartListener(editorChangeListener);
 			}
 		});
+
+		// ProjectMetadata projMeta = new ProjectMetadata();
+		// projMeta.setName(StringConstants.PROJ_NAME);
+		// projMeta.setProjectId(StringConstants.PROJ_ID);
+		// projMeta.setOwner(StringConstants.PREFERENCES_USERNAME);
+		//
+		// FileMetadata fileMeta = new FileMetadata();
+		// fileMeta.setFileId(StringConstants.FILE_ID);
+		// fileMeta.setFilePath(StringConstants.FILE_PATH);
+		// fileMeta.setProjectId(StringConstants.PROJ_ID);
+		// fileMeta.setVersion(5);
+		//
+		// projMeta.setFiles(new FileMetadata[] { fileMeta });
+		//
+		// metadataManager.writeProjectMetadata(projMeta,
+		// ResourcesPlugin.getWorkspace().getRoot().getLocation().toString());
+
+		System.out.println("PROJECT_METADATA: " + metadataManager
+				.getProjectMetadata(ResourcesPlugin.getWorkspace().getRoot().getLocation().toString() + "/"));
 	}
 
 	public WSManager getWSManager() {
 		return wsManager;
 	}
 
+	public DocumentManager getDocumentManager() {
+		return documentManager;
+	}
+
+	public MetadataManager getMetadataManager() {
+		return metadataManager;
+	}
+
+	public DataManager getDataManager() {
+		return dataManager;
+	}
+
 	private void registerNotificationHooks() {
 		wsManager.registerNotificationHandler("File", "Change",
-				(Notification n) -> DocumentManager.getInstance().handleNotification(n));
+				(Notification n) -> documentManager.handleNotification(n));
 	}
 
+	private void websocketLogin() throws ConnectException, InterruptedException {
+		Request req1 = new UserRegisterRequest(StringConstants.PREFERENCES_USERNAME,
+				StringConstants.PREFERENCES_FIRSTNAME, StringConstants.PREFERENCES_LASTNAME,
+				StringConstants.PREFERENCES_EMAIL, StringConstants.PREFERENCES_PASSWORD).getRequest(resp -> {
+
+					Request req2 = new UserLoginRequest(StringConstants.PREFERENCES_USERNAME,
+							StringConstants.PREFERENCES_PASSWORD).getRequest(response -> {
+								// TODO(wongb) Add login logic for server
+								if (response.getStatus() != 200) {
+									throw new IllegalStateException("Failed to log in");
+								}
+
+								getWSManager().setAuthInfo(StringConstants.PREFERENCES_USERNAME,
+										((UserLoginResponse) response.getData()).getToken());
+
+								try {
+									wsManager.sendRequest(new ProjectSubscribeRequest(StringConstants.PROJ_ID)
+											.getRequest(null, null));
```

Comment:
Ok, just making sure. I've got a couple of classes that are just generic response and error handlers that open up dialogs, so it should be a really simple change once those are merged.